2020/dec17/energysource/source4d/state.go

```diff
@@ -0,0 +1,64 @@
+package source4d
+
+// State is an implementation of the Experimental Energy Source
+// It represents the state of the Source at a given point in the iterative dimension (think time, for example)
+type State struct {
+	active map[coord4D]struct{}
+	min    coord4D
+	max    coord4D
```

Comment:
these two were only used for printing, and source4d.State doesn't have a String() method.

Reply:
They are used in `iterate()`.

Reply:
ah, very true, missed them.

This makes me wonder what the density of cubes is.